lxd/instance.go

```diff
@@ -416,7 +416,68 @@ func instanceLoadNodeProjectAll(ctx context.Context, s *state.State, project str
 	return instances, nil
 }
 
-func autoCreateInstanceSnapshotsTask(d *Daemon) (task.Func, task.Schedule) {
+func autoCreateInstanceSnapshots(ctx context.Context, d *Daemon, instances []instance.Instance) error {
+	// Make the snapshots.
+	for _, inst := range instances {
+		ch := make(chan error)
+		go func(inst instance.Instance) {
+			l := logger.AddContext(logger.Log, logger.Ctx{"project": inst.Project().Name, "instance": inst.Name()})
+
+			snapshotName, err := instance.NextSnapshotName(d.State(), inst, "snap%d")
+			if err != nil {
+				l.Error("Error retrieving next snapshot name", logger.Ctx{"err": err})
+				ch <- nil
+				return
+			}
+
+			expiry, err := shared.GetExpiry(time.Now(), inst.ExpandedConfig()["snapshots.expiry"])
+			if err != nil {
+				l.Error("Error getting snapshots.expiry date")
+				ch <- nil
+				return
+			}
+
+			err = inst.Snapshot(snapshotName, expiry, false)
+			if err != nil {
+				l.Error("Error creating snapshot", logger.Ctx{"snapshot": snapshotName, "err": err})
+			}
+
+			ch <- nil
+		}(inst)
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-ch:
+		}
+	}
+
+	return nil
+}
+
+var instSnapshotsPruneRunning = sync.Map{}
+
+func pruneExpiredInstanceSnapshots(ctx context.Context, d *Daemon, snapshots []instance.Instance) error {
```

Comment:
Was this function changed or just moved?

Reply:
just moved

Reply:
OK, so in the future, to make it easier to review, you can split that into 2 commits or alternatively leave it where it was.